### PR TITLE
Integrate with Travis CI and the E3SM singularity container

### PR DESCRIPTION
**Pull request by lukaszlacinski**
The PR integrates the E3SM repository with Travis CI. The .travis.yml file describes a Travis CI deployment environment and how to clone the E3SM repository with all submodules. It also tells Travis CI, that after the environment is set up, to run
1) .travis/setup.sh script to install Go and Singularity,
2) .travis/run.sh script to create, set up, and build A_WCYCL1850.ne4_oQU240 case
Step 2) takes from 5 to 15 minutes on the Travis CI platform. Such a wide range of execution time can be caused by different total compute load and memory consumption on a machine where a VM for the deployment environment is launched.

[BFB]

### CODE REVIEW

File: .travis/run.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        exit $rc
+    fi
+}
+
+command_yellow_rc "mkdir -p $HOME/projects/e3sm/cesm-inputdata"
+command_yellow_rc 'cd cime/scripts'
+command_yellow_rc './create_newcase --case master.A_WCYCL1850.ne4_oQU240.baseline --compset A_WCYCL1850 --res ne4_oQU240'
+command_yellow_rc 'cd master.A_WCYCL1850.ne4_oQU240.baseline'
+command_yellow_rc './case.setup'
+command_yellow './case.build'
```
Comment (rljacob) [inline .travis/run.sh:25]: So you are only trying to build and not run?  If the case, I think we can avoid downloading data.

Comment (lukaszlacinski) [reply .travis/run.sh:25]: If I am not mistaken any missing input data are downloaded in the "run" step.

Comment (rljacob) [reply .travis/run.sh:25]: Oh yeah that's right.  Good.

**Comment by rljacob** [comment]
What is the status of this PR?  It is still "draft".

**Review by rljacob — APPROVED** [review]

**Pull request merged**